### PR TITLE
nqc: init at 3.1.r6

### PR DESCRIPTION
**Pull request by christophcharles**
###### Description of changes

Added package nqc ([https://bricxcc.sourceforge.net/nqc/](https://bricxcc.sourceforge.net/nqc/))

NQC is a compiler for the NQC language, a programming language for several lego mindstorms products including the RCX, CyberMaster, and Scout.

The compiler is quite old. I have included two patches:
- one from https://sourceforge.net/p/bricxcc/patches/_discuss/thread/00b427dc/b84b/attachment/nqc-01-Linux_usb_and_tcp.diff which is needed to use the modern kernel driver for the usb lego tower.
- a small one adding an include of unistd.h to solve some compilation issues when using nix.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/nqc/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "nqc";
+  version = "3.1.r6";
+
+  src = fetchurl {
+    url    = "https://bricxcc.sourceforge.net/nqc/release/nqc-${version}.tgz";
```
Comment (azahi) [inline pkgs/development/compilers/nqc/default.nix:8]:
```suggestion
    url = "https://bricxcc.sourceforge.net/nqc/release/nqc-${version}.tgz";
```

File: pkgs/development/compilers/nqc/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "sha256-v9XmVPY5r3pYjP3vTSK9Xvz/9UexClbOvr3ljvK/52Y=";
+  };
+
+  sourceRoot = ".";
+
+  patches = [
+    ./nqc-unistd.patch
+    (fetchurl {
```
Comment (azahi) [inline pkgs/development/compilers/nqc/default.nix:16]:
```suggestion
    (fetchpatch {
```

**Review by azahi — CHANGES_REQUESTED** [review]
Please split changes to the `maintainer-list.nix` into a separate commit.

**Comment by christophcharles** [comment]
Thanks @azahi for reviewing.
I've added the proposed changes and split the commit into two.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 196026` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nqc</li>
  </ul>
</details>

**Comment by christophcharles** [comment]
Sorry, I must have pushed the wrong one. I've run `nixpkgs-review` to check other silly mistakes.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 196026` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nqc</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Comment by viraptor** [comment]
Can't test with hardware, but it seems to start otherwise.

**Pull request merged**